### ISSUE TITLE
pypy3Packages.bootstrapped-pip fails to build

### ISSUE THREAD

**Issue by wamserma** [NixOS/nixpkgs#150818]
### Describe the bug
`nix-build -A pypy3Packages.bootstrapped-pip https://github.com/nixos/nixpkgs/archive/977810ca505c4aed35e99f11de0d3d86a0e6c425.tar.gz` fails when building the wheel for pip. (Building the wheel for wheel succeeds.)

### Steps To Reproduce
1. `nix-build -A pypy3Packages.bootstrapped-pip https://github.com/nixos/nixpkgs/archive/977810ca505c4aed35e99f11de0d3d86a0e6c425.tar.gz`

### Expected behavior
Build succeeds as it does for regular Python:
`nix-build -A python3Packages.bootstrapped-pip https://github.com/nixos/nixpkgs/archive/977810ca505c4aed35e99f11de0d3d86a0e6c425.tar.gz`

### Notify maintainers
@FRidh @domenkozar (there are no maintainers given)

**Comment by FRidh** [comment]
Which branch is this on?

**Comment by FRidh** [comment]
Can't reproduce this on master nor on nixos-22.05.

**Comment by FRidh** [comment]
```sh
$ nix-build -A pypy3Packages.bootstrapped-pip
this derivation will be built:
  /nix/store/swshh82vrqkfsdqg9mskdipy0fxjpc1a-pypy3.8-bootstrapped-pip-22.0.4.drv
these 4 paths will be fetched (24.18 MiB download, 132.93 MiB unpacked):
  /nix/store/1h5g5p34qp1v2m1sxm0lqkq6xibqxpsp-pip-install-hook
  /nix/store/61mkqrly3l3d0wg3knjmnd3rjanla2ad-pypy3-7.3.7
  /nix/store/6xlx0fqik037780jwbvqh7ri0is62g9b-setuptools-61.2.0-sdist.tar.gz
  /nix/store/8jfy19g5y6nmz2wwgjh1xzsancs7i4k2-setuptools-setup-hook
copying path '/nix/store/6xlx0fqik037780jwbvqh7ri0is62g9b-setuptools-61.2.0-sdist.tar.gz' from 'https://cache.nixos.org'...
copying path '/nix/store/61mkqrly3l3d0wg3knjmnd3rjanla2ad-pypy3-7.3.7' from 'https://cache.nixos.org'...
copying path '/nix/store/8jfy19g5y6nmz2wwgjh1xzsancs7i4k2-setuptools-setup-hook' from 'https://cache.nixos.org'...
copying path '/nix/store/1h5g5p34qp1v2m1sxm0lqkq6xibqxpsp-pip-install-hook' from 'https://cache.nixos.org'...
building '/nix/store/swshh82vrqkfsdqg9mskdipy0fxjpc1a-pypy3.8-bootstrapped-pip-22.0.4.drv'...
Sourcing pip-install-hook
Sourcing setuptools-build-hook
Using setuptoolsShellHook
unpacking sources
unpacking source archive /nix/store/3dki8ikais8yg1npyrznzd39vpg66d79-wheel-0.37.1-source
unpacking source archive /nix/store/ry8d2lgffkbf4dn7rqv4snlcxakgrm15-pip-22.0.4-source
unpacking source archive /nix/store/6xlx0fqik037780jwbvqh7ri0is62g9b-setuptools-61.2.0-sdist.tar.gz
source root is .
patching sources
configuring
no configure script, doing nothing
installing
Building setuptools wheel...
/build/setuptools /build
Processing /build/setuptools
  Preparing metadata (setup.py) ... done
Using legacy 'setup.py install' for setuptools, since package 'wheel' is not installed.
Installing collected packages: setuptools
  Running setup.py install for setuptools ... done
Successfully installed setuptools-61.2.0.post0
/build
Building wheel wheel...
/build/wheel /build
Processing /build/wheel
  Preparing metadata (setup.py) ... done
Using legacy 'setup.py install' for wheel, since package 'wheel' is not installed.
Installing collected packages: wheel
  Running setup.py install for wheel ... done
Successfully installed wheel-0.37.1
/build
Building pip wheel...
/build/pip /build
Processing /build/pip
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  Preparing metadata (pyproject.toml) did not run successfully.
  exit code: 1
  
  [31 lines of output]
  running dist_info
  creating /build/pip-modern-metadata-9m64ue0z/pip.egg-info
  writing /build/pip-modern-metadata-9m64ue0z/pip.egg-info/PKG-INFO
  writing dependency_links to /build/pip-modern-metadata-9m64ue0z/pip.egg-info/dependency_links.txt
  writing entry points to /build/pip-modern-metadata-9m64ue0z/pip.egg-info/entry_points.txt
  writing top-level names to /build/pip-modern-metadata-9m64ue0z/pip.egg-info/top_level.txt
  writing manifest file '/build/pip-modern-metadata-9m64ue0z/pip.egg-info/SOURCES.txt'
  reading manifest file '/build/pip-modern-metadata-9m64ue0z/pip.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  warning: no files found matching 'docs/docutils.conf'
  warning: no previously-included files found matching '.coveragerc'
  warning: no previously-included files found matching '.mailmap'
  warning: no previously-included files found matching '.appveyor.yml'
  warning: no previously-included files found matching '.readthedocs.yml'
  warning: no previously-included files found matching '.pre-commit-config.yaml'
  warning: no previously-included files found matching 'tox.ini'
  warning: no previously-included files found matching 'noxfile.py'
  warning: no files found matching '*.css' under directory 'docs'
  warning: no previously-included files found matching 'src/pip/_vendor/six'
  warning: no previously-included files found matching 'src/pip/_vendor/six/moves'
  warning: no previously-included files matching '*.pyi' found under directory 'src/pip/_vendor'
  no previously-included directories found matching '.github'
  no previously-included directories found matching 'docs/build'
  no previously-included directories found matching 'news'
  no previously-included directories found matching 'tasks'
  no previously-included directories found matching 'tests'
  no previously-included directories found matching 'tools'
  adding license file 'LICENSE.txt'
  writing manifest file '/build/pip-modern-metadata-9m64ue0z/pip.egg-info/SOURCES.txt'
  creating '/build/pip-modern-metadata-9m64ue0z/pip.dist-info'
  error: invalid command 'bdist_wheel'
  [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

Encountered error while generating package metadata.

See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details
```

**Comment by ShamrockLee** [comment]
Encountered on Nixpkgs 21.11

 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.156, NixOS, 22.11 (Raccoon), 22.11.20221130.596a8e8`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - nixpkgs: `/nix/store/qwgw2b19ga8dr7c6wfpd6p87bdjyrnha-source`

```console
❯ nix build -L nixpkgs-local#pypy3Packages.weasyprint
pypy3.9-bootstrapped-pip> Sourcing pip-install-hook
pypy3.9-bootstrapped-pip> Sourcing setuptools-build-hook
pypy3.9-bootstrapped-pip> Using setuptoolsShellHook
pypy3.9-bootstrapped-pip> unpacking sources
pypy3.9-bootstrapped-pip> unpacking source archive /nix/store/3dki8ikais8yg1npyrznzd39vpg66d79-wheel-0.37.1-source
pypy3.9-bootstrapped-pip> unpacking source archive /nix/store/a859qkahn9df557sbvi4nl18clpqdwsa-pip-22.2.2-source
pypy3.9-bootstrapped-pip> unpacking source archive /nix/store/jab7chnlsciyjln2h479m5n7swbai8wd-setuptools-65.3.0-sdist.tar.gz
pypy3.9-bootstrapped-pip> source root is .
pypy3.9-bootstrapped-pip> patching sources
pypy3.9-bootstrapped-pip> configuring
pypy3.9-bootstrapped-pip> no configure script, doing nothing
pypy3.9-bootstrapped-pip> installing
pypy3.9-bootstrapped-pip> Building setuptools wheel...
pypy3.9-bootstrapped-pip> /build/setuptools /build
pypy3.9-bootstrapped-pip> Processing /build/setuptools
pypy3.9-bootstrapped-pip>   Preparing metadata (setup.py) ... done
pypy3.9-bootstrapped-pip> Using legacy 'setup.py install' for setuptools, since package 'wheel' is not installed.
pypy3.9-bootstrapped-pip> Installing collected packages: setuptools
pypy3.9-bootstrapped-pip>   Running setup.py install for setuptools ... done
pypy3.9-bootstrapped-pip> Successfully installed setuptools-65.3.0.post0
pypy3.9-bootstrapped-pip> /build
pypy3.9-bootstrapped-pip> Building wheel wheel...
pypy3.9-bootstrapped-pip> /build/wheel /build
pypy3.9-bootstrapped-pip> Processing /build/wheel
pypy3.9-bootstrapped-pip>   Preparing metadata (setup.py) ... done
pypy3.9-bootstrapped-pip> Using legacy 'setup.py install' for wheel, since package 'wheel' is not installed.
pypy3.9-bootstrapped-pip> Installing collected packages: wheel
pypy3.9-bootstrapped-pip>   Running setup.py install for wheel ... done
pypy3.9-bootstrapped-pip> Successfully installed wheel-0.37.1
pypy3.9-bootstrapped-pip> /build
pypy3.9-bootstrapped-pip> Building pip wheel...
pypy3.9-bootstrapped-pip> /build/pip /build
pypy3.9-bootstrapped-pip> Processing /build/pip
pypy3.9-bootstrapped-pip>   Preparing metadata (pyproject.toml) ... error
pypy3.9-bootstrapped-pip>   error: subprocess-exited-with-error
pypy3.9-bootstrapped-pip>   × Preparing metadata (pyproject.toml) did not run successfully.
pypy3.9-bootstrapped-pip>   │ exit code: 1
pypy3.9-bootstrapped-pip>   ╰─> [34 lines of output]
pypy3.9-bootstrapped-pip>       running dist_info
pypy3.9-bootstrapped-pip>       creating /build/pip-modern-metadata-5suy2vvr/pip.egg-info
pypy3.9-bootstrapped-pip>       writing /build/pip-modern-metadata-5suy2vvr/pip.egg-info/PKG-INFO
pypy3.9-bootstrapped-pip>       writing dependency_links to /build/pip-modern-metadata-5suy2vvr/pip.egg-info/dependency_links.txt
pypy3.9-bootstrapped-pip>       writing entry points to /build/pip-modern-metadata-5suy2vvr/pip.egg-info/entry_points.txt
pypy3.9-bootstrapped-pip>       writing top-level names to /build/pip-modern-metadata-5suy2vvr/pip.egg-info/top_level.txt
pypy3.9-bootstrapped-pip>       writing manifest file '/build/pip-modern-metadata-5suy2vvr/pip.egg-info/SOURCES.txt'
pypy3.9-bootstrapped-pip>       reading manifest file '/build/pip-modern-metadata-5suy2vvr/pip.egg-info/SOURCES.txt'
pypy3.9-bootstrapped-pip>       reading manifest template 'MANIFEST.in'
pypy3.9-bootstrapped-pip>       /build/setuptools/setuptools/config/setupcfg.py:508: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
pypy3.9-bootstrapped-pip>         warnings.warn(msg, warning_class)
pypy3.9-bootstrapped-pip>       warning: no files found matching 'src/pip/_vendor/pyparsing/diagram/template.jinja2'
pypy3.9-bootstrapped-pip>       warning: no files found matching 'docs/docutils.conf'
pypy3.9-bootstrapped-pip>       warning: no previously-included files found matching '.coveragerc'
pypy3.9-bootstrapped-pip>       warning: no previously-included files found matching '.mailmap'
pypy3.9-bootstrapped-pip>       warning: no previously-included files found matching '.appveyor.yml'
pypy3.9-bootstrapped-pip>       warning: no previously-included files found matching '.readthedocs.yml'
pypy3.9-bootstrapped-pip>       warning: no previously-included files found matching '.pre-commit-config.yaml'
pypy3.9-bootstrapped-pip>       warning: no previously-included files found matching 'tox.ini'
pypy3.9-bootstrapped-pip>       warning: no previously-included files found matching 'noxfile.py'
pypy3.9-bootstrapped-pip>       warning: no files found matching '*.css' under directory 'docs'
pypy3.9-bootstrapped-pip>       warning: no previously-included files found matching 'src/pip/_vendor/six'
pypy3.9-bootstrapped-pip>       warning: no previously-included files found matching 'src/pip/_vendor/six/moves'
pypy3.9-bootstrapped-pip>       warning: no previously-included files matching '*.pyi' found under directory 'src/pip/_vendor'
pypy3.9-bootstrapped-pip>       no previously-included directories found matching '.github'
pypy3.9-bootstrapped-pip>       no previously-included directories found matching 'docs/build'
pypy3.9-bootstrapped-pip>       no previously-included directories found matching 'news'
pypy3.9-bootstrapped-pip>       no previously-included directories found matching 'tasks'
pypy3.9-bootstrapped-pip>       no previously-included directories found matching 'tests'
pypy3.9-bootstrapped-pip>       no previously-included directories found matching 'tools'
pypy3.9-bootstrapped-pip>       adding license file 'LICENSE.txt'
pypy3.9-bootstrapped-pip>       writing manifest file '/build/pip-modern-metadata-5suy2vvr/pip.egg-info/SOURCES.txt'
pypy3.9-bootstrapped-pip>       creating '/build/pip-modern-metadata-5suy2vvr/pip-22.2.2.dist-info'
pypy3.9-bootstrapped-pip>       error: invalid command 'bdist_wheel'
pypy3.9-bootstrapped-pip>       [end of output]
pypy3.9-bootstrapped-pip>   note: This error originates from a subprocess, and is likely not a problem with pip.
pypy3.9-bootstrapped-pip> error: metadata-generation-failed
pypy3.9-bootstrapped-pip> × Encountered error while generating package metadata.
pypy3.9-bootstrapped-pip> ╰─> See above for output.
pypy3.9-bootstrapped-pip> note: This is an issue with the package mentioned above, not pip.
pypy3.9-bootstrapped-pip> hint: See above for details.
pypy3.9-bootstrapped-pip>
error: builder for '/nix/store/iz6kj4jdw5gh6sf3f4a05jz36lfxypxh-pypy3.9-bootstrapped-pip-22.2.2.drv' failed with exit code 1;
       last 10 log lines:
       >
       >   note: This error originates from a subprocess, and is likely not a problem with pip.
       > error: metadata-generation-failed
       >
       > × Encountered error while generating package metadata.
       > ╰─> See above for output.
       >
       > note: This is an issue with the package mentioned above, not pip.
       > hint: See above for details.
       >
       For full logs, run 'nix log /nix/store/iz6kj4jdw5gh6sf3f4a05jz36lfxypxh-pypy3.9-bootstrapped-pip-22.2.2.drv'.
error: 1 dependencies of derivation '/nix/store/yakgvc3g96gkn412qfaa3x7zvyi9z1vf-pypy3.9-pip-22.2.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/jwcyapk5fyrnz8awrfvdy3fh7sghs7p7-pypy3.9-setuptools-65.3.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/p9ap54d1r60dwqghdc1p19k9nw0kfi7s-pip-build-hook.sh.drv' failed to build
error: 1 dependencies of derivation '/nix/store/xxgkcbv3ja4p4mj432xxv62a9r3p7bib-pip-install-hook.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ziv0va20mhbb4rpsxsw6x883v6hds0mv-pypy3.9-packaging-21.3.drv' failed to build
error: 1 dependencies of derivation '/nix/store/8pp10fcbdb518shfpxgrmw2v49631sxx-pypy3.9-pathspec-0.10.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/hi0r81fqvs18d9d393192jbyl44rzbn8-pypy3.9-pdm-pep517-1.0.4.drv' failed to build
error: 1 dependencies of derivation '/nix/store/3k0i5aihmyd1sxijb887xkxqlx0f84gr-pypy3.9-pyroma-3.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/inadazxgczgqvmw9l0v1n57v3kk9gv1z-pypy3.9-setuptools-scm-7.0.5.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5mr3iq8jbsly1q2a00c8lix6gap8sfx4-pypy3.9-skia-pathops-0.7.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/rgycwmkcv6la12ckpbg0z4xw686slzyc-python-catch-conflicts-hook.drv' failed to build
error: 1 dependencies of derivation '/nix/store/zrmyh3dbx7vqwilim0zdkxwh74qsgny9-setuptools-check-hook.drv' failed to build
error: 1 dependencies of derivation '/nix/store/s4wq274w3q95mdk5lsgmcq3cqi4gqp0w-setuptools-setup-hook.drv' failed to build
error: 1 dependencies of derivation '/nix/store/x9sxs42kx2f3l8nahs1c7xz61vzgb721-pypy3.9-weasyprint-57.0.drv' failed to build
```

**Comment by mweinelt** [comment]
Bisected to the 7.3.7 update. Newer pypy3 versions exhibit the same problem. Our pypy3 package set is clearly unmaintained.

```
eec28b8cfd7923a876fb47af8a0459c1a550fd97 is the first bad commit
commit eec28b8cfd7923a876fb47af8a0459c1a550fd97
Author: Sumner Evans <me@sumnerevans.com>
Date:   Sun Dec 5 12:21:35 2021 -0700

    pypy3: 7.3.5 -> 7.3.7 (#147875)
    
    This adds support for Python 3.8.
    https://www.pypy.org/posts/2021/10/pypy-v737-release.html

 pkgs/development/interpreters/python/default.nix | 10 +++++-----
 pkgs/top-level/all-packages.nix                  |  4 ++-
```

https://github.com/NixOS/nixpkgs/pull/147875

cc @sumnerevans

**Comment by winterqt** [comment]
Fixed in #217558.
